hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java

```diff
@@ -111,12 +127,51 @@ public interface PreHandleContext {
     @NonNull
     PreHandleContext requireKey(@NonNull final Key key);
 
+    /**
+     * Adds the given key to optional non-payer keys.
+     * If the key is the same as the payer key, or if the key has already been added, then the call
```

Comment:
```suggestion
     * If the key is invalid, is the same as the payer key, or if the key has already been added, then the call
```

Unless it's necessary to allow invalid keys, I suggest that the key param's conditions of validity be established in the API documentation (as opposed to an implementation detail) so we don't have to worry about invalid cases separately in each implementation

Reply:
We do have to allow invalid keys as preHandle doesn't always have the state necessary to verify that keys are valid, and checking validity requires quite a bit of work that shouldn't be done until Signature Verification.  Also, for Optional items, invalid is definitely allowed.  The decision of whether an invalid key is permitted is left to the Handler that added the optional item in that case.

Reply:
Will fix.
Thanks.